### PR TITLE
Add TargetIota objective

### PR DESCRIPTION
**Pull request by f0uriest**
Bringing this back from the dead. Now that we can compute with fixed current, it's useful to be able to target a particular iota profile.

### CODE REVIEW

File: desc/objectives/_generic.py [hunk 1]
```diff
@@ -320,4 +331,152 @@ def compute(self, R_lmn, Z_lmn, L_lmn, i_l, c_l, Psi, **kwargs):
...
+        Note: has no effect for this objective.
+    grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at.
+    name : str
+        Name of the objective function.
+    """
+
+    _scalar = False
+    _linear = False
+    _units = "(n/a)"
```
Comment (ddudt) [inline desc/objectives/_generic.py:366]: In other objectives (like aspect ratio) we say `"(dimensionless)"`. We should be consistent -- which do we prefer?

File: tests/test_objective_funs.py [hunk 2]
```diff
@@ -99,14 +100,26 @@ def test(eq):
...
         test(Equilibrium(node_pattern="quad", current=PowerSeriesProfile(0)))
 
+    @pytest.mark.unit
+    def test_target_iota(self):
+        """Test calculation of iota profile."""
+
+        def test(eq):
+            obj = TargetIota(target=1, weight=2, eq=eq)
+            iota = obj.compute(eq.R_lmn, eq.Z_lmn, eq.L_lmn, eq.i_l, eq.c_l, eq.Psi)
+            np.testing.assert_allclose(iota, -2 / 3)
```
Comment (ddudt) [inline tests/test_objective_funs.py:110]: Actual iota is 0 but the target is 1, so the error is -1. Then with a weight of 2 I would expect the result to be -2. How do we get -2/3?

**Review by ddudt — DISMISSED** [review]

File: tests/test_objective_funs.py (hunk 2 above)
Comment (f0uriest) [reply tests/test_objective_funs.py:110]: This is the grid spacing thing I mentioned. The residual is `(f-target)*weight*spacing` and in this case the default grid has `L_grid` radial nodes (and `L_grid` is 3

File: desc/objectives/_generic.py (hunk 1 above)
Comment (f0uriest) [reply desc/objectives/_generic.py:366]: yeah dimensionless is probably better ill change it

File: tests/test_objective_funs.py (hunk 2 above)
Comment (ddudt) [reply tests/test_objective_funs.py:110]: Ah OK. Sounds good.

**Review by ddudt — APPROVED** [review]

**Review by dpanici — APPROVED** [review]

**Pull request merged**